Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -838,6 +1091,290 @@ void MatchTOF::doMatching(int sec)
   return;
 }
 //______________________________________________
+void MatchTOF::doMatchingForTPC(int sec)
+{
+
+  auto& gasParam = o2::tpc::ParameterGas::Instance();
+  float vdrift = gasParam.DriftV;
+
+  ///< do the real matching per sector
+  mMatchedTracksPairs.clear(); // new sector
+
+  auto& cacheTOF = mTOFClusSectIndexCache[sec]; // array of cached TOF cluster indices for this sector; reminder: they are ordered in time!
+  auto& cacheTrk = mTracksSectIndexCache[sec];  // array of cached tracks indices for this sector; reminder: they are ordered in time!
+  int nTracks = cacheTrk.size(), nTOFCls = cacheTOF.size();
+  LOG(INFO) << "Matching sector " << sec << ": number of tracks: " << nTracks << ", number of TOF clusters: " << nTOFCls;
+  if (!nTracks || !nTOFCls) {
+    return;
+  }
+  int itof0 = 0;                          // starting index in TOF clusters for matching of the track
+  int detId[2][5];                        // at maximum one track can fall in 2 strips during the propagation; the second dimention of the array is the TOF det index
+  float deltaPos[2][3];                   // at maximum one track can fall in 2 strips during the propagation; the second dimention of the array is the residuals
+  o2::track::TrackLTIntegral trkLTInt[2]; // Here we store the integrated track length and time for the (max 2) matched strips
+  int nStepsInsideSameStrip[2] = {0, 0};  // number of propagation steps in the same strip (since we have maximum 2 strips, it has dimention = 2)
+  float deltaPosTemp[3];
+  std::array<float, 3> pos;
+  std::array<float, 3> posBeforeProp;
+  float posFloat[3];
+
+  // prematching for TPC only tracks (identify BC candidate to correct z for TPC track accordingly to v_drift)
+
+  LOG(DEBUG) << "Trying to match %d tracks" << cacheTrk.size();
+  for (int itrk = 0; itrk < cacheTrk.size(); itrk++) {
+    int nStripsCrossedInPropagation = 0; // how many strips were hit during the propagation
+    auto& trackWork = mTracksWork[cacheTrk[itrk]];
+    auto& trefTrk = trackWork.first;
+    auto& intLT = mLTinfos[cacheTrk[itrk]];
+
+    int nBCcand = 1;
+    std::vector<ulong> BCcand;
+    BCcand.reserve(nBCcand);
```

Comment:
better to define this vector above the track loop, and do ``bcCand.clear()``, to not allocate the memory for every iteration

Reply:
Done